fillpdf.go

```diff
@@ -32,9 +32,17 @@ import (
 // This is a key value map.
 type Form map[string]interface{}
 
+// Options represents the options to alter the PDF filling process
+// Overwrite will overwrite any pre existing filled PDF
+// Flatten will flatten the document making the form field no longer editable
+type Options struct {
+	Overwrite bool
+	Flatten   bool
+}
+
 // Fill a PDF form with the specified form values and create a final filled PDF file.
-// One variadic boolean specifies, whenever to overwrite the destination file if it exists.
-func Fill(form Form, formPDFFile, destPDFFile string, overwrite ...bool) (err error) {
+// The options parameter alters few aspects of the generation.
+func Fill(form Form, formPDFFile, destPDFFile string, options Options) (err error) {
```

Comment:
Make options variadic, to keep in sync with the previous API.

```go
var o Options
if len(options) > 0 {
  o = options[0]
}
```

Reply:
Implemented default values and made the parameter variadic again.